### PR TITLE
PRs by author

### PR DESCRIPTION
**Pull request by tobyselway**

### CODE REVIEW

File: queries/author/author.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+	for _, node := range result.Nodes {
+
+		pr := gitter.PullRequest{
+			Title:          node.Title,
+			BaseRefName:    node.BaseRefName,
+			HeadRefName:    node.HeadRefName,
+			Number:         node.Number,
+			Permalink:      node.Permalink,
+			ReviewCount:    0, // TODO
```
Comment (DarVoid) [inline queries/author/author.go:46]: we should address this asap, I think there's potential here for a nice feature like: 3/4 people who reviewed approved or sth like that

File: queries/author/author.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		all = append(all, pr)
+	}
+
+	return all
+}
+
+func (query Query) GetAll(client gitter.GraphQLClient) (map[string][]gitter.PullRequest, error) {
+	res := Result{}
+
+	if strings.Trim(query.generatedQuery, "\n") != "" {
```
Comment (DarVoid) [inline queries/author/author.go:61]: else log error empty query

File: queries/author/author.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		err := gitter.AuthorizedGraphQLQuery[Result](client, query.generatedQuery, &res)
+		if err != nil {
+			return map[string][]gitter.PullRequest{}, fmt.Errorf("requesting PRs: %w", err)
```
Comment (DarVoid) [inline queries/author/author.go:64]: in case of error this won't say much

File: queries/author/author.go (hunk 2 above)
Comment (tobyselway) [reply queries/author/author.go:61]: This only happens if there are no trackers registered of that type, it's not an error

File: queries/author/author.go (hunk 3 above)
Comment (DarVoid) [inline queries/author/author.go:64]:
```suggestion
			return map[string][]gitter.PullRequest{}, fmt.Errorf("author response failed: %w", err)
```
or sth like that

File: queries/author/author_test.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			name: "ReviewAmount > 0",
+			inputs: author.Config{
+				Trackers:       []author.Tracker{},
+				ReviewAmount:   0,
+				PrAmount:       10,
+				CommentsAmount: 10,
+			},
+			expectedError: errors.New("invalid config: ReviewAmount must not be zero or less"),
+		},
+		// TODO: add more cases
```
Comment (DarVoid) [inline queries/author/author_test.go:48]: non empty authors maybe

File: queries/author/author.go (hunk 3 above)
Comment (tobyselway) [reply queries/author/author.go:64]: how so? it wraps the err returned from the graphql client, isn't that enough context?

File: queries/author/initialize.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	if err != nil {
+		return Query{}, fmt.Errorf("error generating query: %w", err)
+	}
+
+	return Query{
+		generatedQuery: query,
+	}, nil
+}
+
+func validateConfig(config Config) error {
```
Comment (DarVoid) [inline queries/author/initialize.go:24]: also here validate len not zero

File: queries/author/author.go (hunk 1 above)
Comment (tobyselway) [reply queries/author/author.go:46]: That's not a small refactor. Added it to the backlog (#35)

File: polls.go [hunk 6]
```diff
@@ -2,15 +2,17 @@ package main
...
+	"log"
+	"os"
+	"time"
+
 	"git_applet/gitter"
 	"git_applet/queries"
 	"git_applet/queries/aggregator"
+	"git_applet/queries/author"
 	"git_applet/queries/labeled"
 	"git_applet/queries/personal"
```
Comment (DarVoid) [inline polls.go:14]: what do you think of changing the packages to by<insertStrategyName>

File: polls.go [hunk 7]
```diff
@@ -74,11 +96,17 @@ func setupPolling(mock bool) error {
 		return fmt.Errorf("setting up polling: %w", err)
 	}
 
+	author, err := setupAuthorQuery()
+	if err != nil {
+		return fmt.Errorf("setting up polling: %w", err)
```
Comment (DarVoid) [inline polls.go:101]:
```suggestion
		return fmt.Errorf("setting up  author polling: %w", err)
```

Comment (DarVoid) [reply polls.go:101]: maybe replicate for others

File: polls.go (hunk 6 above)
Comment (tobyselway) [reply polls.go:14]: `by_author`, `by_label`, `by_repo`, `personal`?

File: queries/author/author.go (hunk 2 above)
Comment (DarVoid) [reply queries/author/author.go:61]: I'd still log it under debug or something

File: queries/author/author.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+		if err != nil {
+			return map[string][]gitter.PullRequest{}, fmt.Errorf("author response failed: %w", err)
+		}
+	}
+
+	prs := make(map[string][]gitter.PullRequest)
+
+	for key, prMap := range res {
+		parts := strings.Split(key, "_")
+		baseKey := strings.Join(parts[0:2], "_")
```
Comment (DarVoid) [inline queries/author/author.go:72]: we're doing this on labeld, repo and authors right? maybe let's create a strings manipulation module?

Comment (tobyselway) [reply queries/author/author.go:72]: Nope, just here. This is to merge the results of multiple queries being ran (one for each author)

File: queries/author/query.go [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+	"github.com/ettle/strcase"
+)
+
+var funcMap = template.FuncMap{
+	"ToSnake": strcase.ToSnake,
+	"ToKebab": strcase.ToKebab,
+	"ToLower": strings.ToLower,
+	"ToUpper": strings.ToUpper,
+}
```
Comment (DarVoid) [inline queries/author/query.go:18]: I believe these could also come from the strings manipulation module (i think our templates are't really using them anyway so they could be deleted)

File: queries/author/author.go (hunk 1 above)
Comment (DarVoid) [reply queries/author/author.go:46]: yeah, this one was not for this PR :p

File: polls.go (hunk 6 above)
Comment (DarVoid) [reply polls.go:14]: yes

**Review by DarVoid — CHANGES_REQUESTED** [review]
good job, keep on it

File: queries/author/author.go (hunk 2 above)
Comment (tobyselway) [reply queries/author/author.go:61]: We can completely avoid this by addressing a bit of hackyness I implemented a while back :sweat_smile:

Comment (DarVoid) [reply queries/author/author.go:61]: ƪ(˘⌣˘)ʃ

**Review by DarVoid — APPROVED** [review]
Ok, now it's good

**Pull request merged**